stdlib/public/core/StringUTF8.swift

```diff
@@ -236,7 +245,13 @@ extension String {
     /// - Precondition: `position` is a valid position in `self` and
     ///   `position != endIndex`.
     public subscript(position: Index) -> UTF8.CodeUnit {
+#if endian(little)
       let result = UTF8.CodeUnit(truncatingBitPattern: position._buffer & 0xFF)
+#else
+      let numBits = (sizeof(_StringCore._UTF8Chunk.self) &- 1) &* 8
+      let hiByte = position._buffer >> numericCast(numBits)
+      let result = UTF8.CodeUnit(truncatingBitPattern: hiByte & 0xFF)
+#endif
```

Comment:
Yes, on big-endian systems the bytes in a `_UTF8Chunk` are laid out in the opposite order. I will try to improve this code.